public/app/features/scopes/selector/ScopesSelectorService.ts

```diff
@@ -244,35 +252,49 @@ export class ScopesSelectorService extends ScopesServiceBase<ScopesSelectorServi
       for (const scope of fetchedScopes) {
         newScopesState[scope.metadata.name] = scope;
       }
-      this.addRecentScopes(fetchedScopes);
+
+      const scopeNode = scopes[0]?.scopeNodeId ? this.state.nodes[scopes[0]?.scopeNodeId] : undefined;
+
+      // If parentNodeId is provided, use it directly as the parent node
+      // If not provided, try to get the parent from the scope node
+      const parentNodeId = scopes[0]?.parentNodeId || scopeNode?.spec.parentName;
```

Comment:
So do we actually need the `parentNodeId` in the argument for this? Like we already have `scopeNodeId` in the `SelectedScope`. If we have that we can get a parentId from there. If we don't have that that means we didn't load scopeNodes (like just updating from the URL) and in that case the UI won't be able to supply `parentNodeId` directly anyway right? Or is there a case where the UI will have `parentNodeId` but not the `scopeNodeId`?

Reply:
The parent is saved to localStorage, then loaded on init. We could do it with the scopeNode too, but that would just be superflous because we don't use it unless you actually open the selector and navigate.

Reply:
But we save parentNode in the local storage anyway. And on init we load the parent node from recent scopes local storage and putting in the cache https://github.com/grafana/grafana/pull/108795/files#diff-3ca801175f4d27a54f96736dc3b62d79ed3ba43beee60fc72a1fc10c1cf91525R75 so still kinda confused when do we deal with only parentId but not the whole node.